lib/SILGen/SILGenLValue.cpp

```diff
@@ -2219,6 +2280,12 @@ static LValue emitLValueForNonMemberVarDecl(SILGenFunction &SGF,
 
     if (address.getType().is<ReferenceStorageType>())
       lv.add<OwnershipComponent>(typeData);
+
+    AbstractionPattern pattern = SGF.SGM.Types.getAbstractionPattern(var);
+    if (pattern.isClangType()) {
```

Comment:
Hang on, should this also check that the types are different? What's the best way to do that? If they happen to be the same we might as well leave the physical component in place.

Reply:
You want to check whether the type-of-rvalue matches the target lowered type.